server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -1772,26 +1783,39 @@ public Boolean providesDerived(Pool pool, String derivedProvidedProductId) {
             }
         }
 
-        return this.getDerivedProvidedProductIdsByPoolIds(poolIds);
+        return this.getDerivedProvidedProductByPoolIds(poolIds, true);
     }
 
     /**
-     * Fetches a mapping of pool IDs to sets of product IDs representing the provided products of
+     * Fetches a mapping of pool IDs to sets of product IDs or UUIDs representing the provided products of
      * the given pool. The returned map will only contain mappings for pools specified in the given
      * collection of pool IDs.
      *
      * @param poolIds
-     *  A collection of pool IDs for which to fetch provided product IDs
+     *  A collection of pool IDs for which to fetch provided product IDs or UUIDs
+     *
+     * @param fetchMode
+     *  Boolean field.
+     *  True  -> Returns mapping of Pool IDs & provided product IDs
+     *  False -> Returns mapping of Pool IDs & provided product UUIDs
      *
      * @return
-     *  A mapping of pool IDs to provided product IDs
+     *  A mapping of pool IDs to provided product IDs or UUIDs
      */
-    public Map<String, Set<String>> getDerivedProvidedProductIdsByPoolIds(Collection<String> poolIds) {
+    public Map<String, Set<String>> getDerivedProvidedProductByPoolIds(Collection<String> poolIds,
```

Comment:
Again, nack on these changes -- we don't want to completely swing a method's operation and output on a single boolean input. Rewrite the method and make changes as necessary.

Reply:
Removed this piece of code from here.